backend/tests/E2E/BookCollection/CreateBookCollectionTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace App\Tests\E2E\BookCollection;
+
+use App\Book\Domain\Entity\Book;
+use App\BookCollection\Application\DTO\BookCollectionDTO;
+use App\BookCollection\Domain\Entity\BookCollection;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\Uid\Uuid;
+
+class CreateBookCollectionTest extends WebTestCase
+{
+    private KernelBrowser $client;
+    private string $id;
+    private string $id2;
+
+    protected function setUp(): void
+    {
+        $this->client = static::createClient();
+        /** @var EntityManagerInterface $manager */
+        $manager = static::getContainer()->get(EntityManagerInterface::class);
+        $book = new Book(
+            Uuid::v4(),
+            'Libro Collana di test',
+            'Autore per collana di test',
+            20.99,
+            'Test inserimento'
+        );
+
+        $manager->persist($book);
+        $this->id = $book->getId()->__toString();
+
+        $book2 = new Book(
+            Uuid::v4(),
+            'Libro 2 Collana di test',
+            'Autore 2 per collana di test',
+            20.99,
+            'Test inserimento'
+        );
+
+        $manager->persist($book2);
+        $this->id2 = $book2->getId()->__toString();
+
+        $bookCollection = BookCollection::newBookCollectionFromDTO(
+            new BookCollectionDTO(
+                'Collana gia presente a db',
+                'Una collana di libri gia presente a db',
+                [$book->getId()->__toString(), $book2->getId()->__toString()]
+            ),
+            [$book, $book2]
+        );
+
+        $manager->persist($bookCollection);
+        $manager->flush();
+    }
+
+    public function testItCreatesNewBookCollection(): void
+    {
+        $headers = ['CONTENT_TYPE' => 'application/json'];
+        $body = json_encode([
+                  'name' => 'Collana di test',
+                  'description' => 'A collection of my favorite books',
+                  'books' => [
+                    $this->id,
+                    $this->id2,
+                  ],
+                ]);
+        self::assertNotFalse($body);
+        $this->client->xmlHttpRequest('POST', '/api/collections', [], [], $headers, $body);
+        $res = $this->client->getResponse();
+        self::assertEquals(201, $res->getStatusCode());
```

Comment:
In alternativa a queste due righe potreste usare anche `$this->assertResponseStatusCodeSame(Response::HTTP_CREATED);`

Reply:
Fixato anche questo!